### PR TITLE
move plugin and rpc code into sdk; other minor tweaks

### PR DESCRIPTION
**Pull request by SwampDragons**
I believe these are the last items that need to move into the SDK.

### CODE REVIEW

File: packer/plugin/plugin_test.go [hunk 1]
```diff
@@ -78,33 +82,41 @@ func TestHelperProcess(*testing.T) {
...
-		fmt.Printf("%s|tcp|:1234\n", APIVersion)
+		fmt.Printf("%s|tcp|:1234\n", pluginsdk.APIVersion)
 		<-make(chan int)
 	case "post-processor":
-		server, err := Server()
+		server, err := pluginsdk.Server()
+		if err != nil {
+			log.Printf("[ERR] %s", err)
+			os.Exit(1)
+		}
```
Comment (azr) [inline packer/plugin/plugin_test.go:92]: Uh !

File: packer/plugin/plugin_test.go [hunk 2]
```diff
@@ -78,33 +82,41 @@ func TestHelperProcess(*testing.T) {
...
+		err = server.RegisterPostProcessor(new(helperPostProcessor))
```
Comment (azr) [inline packer/plugin/plugin_test.go:93]: Uh ! Unchecked error !

**Review by azr — APPROVED** [review]
I think this is a great PR !

**Comment by azr** [comment]
My #10272 PR wants to move things into the new plugin folder that was introduced by this PR. I think it is safe to merge this to unblock my PR as this PR feels super safe as it moves things around.

File: packer/plugin/plugin_test.go (hunk 2 above)
Comment (SwampDragons) [reply packer/plugin/plugin_test.go:93]: Linter got mad at me. I doubt it's necessary, but it was easier to go with the flow.

**Pull request merged**